src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import logging
+import numpy as np
+from nni.assessor import Assessor, AssessResult
+from .modelfactory import CurveModel
+
+logger = logging.getLogger('curvefitting_Assessor')
+THRESHOLD = 0.95
+
+class CurvefittingAssessor(Assessor):
+    '''
+    CurvefittingAssessor uses learning curve fitting algorithm to predict the learning curve performance in the future.
+    It stops a pending trial X at step S if the trial's forecast result at target step is convergence and lower than the best performance in the history.
+    '''
+    def __init__(self, start_step=6, epoch_num=20):
+        if not start_step > 0:
+            logger.warning('start step should be a positive number')
+        if not epoch_num > 0:
+            logger.warning('number of epoch should be a positive number')
```

Comment:
if epoch_num equals 0, what is the behavior of this assessor?

Reply:
Epoch_num can't be 0. When user writes config file, epoch_num is required instead of optional, and the range is between 1 - 9999